src/react-components/src/__tests__/components/SearchBar-Test.tsx

```diff
@@ -0,0 +1,41 @@
+import { expect } from "chai";
+import { stub } from "sinon";
+import * as Enzyme from "enzyme";
+import * as React from "react";
+
+import * as Mocha from "mocha";
+
+import SearchBar from "../../components/02-molecules/SearchBar/SearchBar";
+
+describe("Search Bar with Optional Parameters", () => {
+  let wrapper: Enzyme.ShallowWrapper<{}, {}>;
+  let searchCallback;
+  let onChangeCallback;
+  beforeEach(() => {
+    searchCallback = stub();
+    onChangeCallback = stub();
+  });
+
+  it("Search Bar has an input field and button", () => {
+    let wrapper = Enzyme.shallow(<SearchBar searchBarId="id" searchButtonId="searchButtonId" searchSubmitHandler={searchCallback}/>);
+    expect(wrapper.find("TextField")).to.have.lengthOf(1);
+    expect(wrapper.find("Button")).to.have.lengthOf(1);
+  });
+
+  it("Search Bar calls callback on Submit ", () => {
+    wrapper = Enzyme.shallow(<SearchBar searchBarId="id" searchButtonId="searchButtonId" searchSubmitHandler={searchCallback}/>);
+    wrapper.find("Button").dive().find("button").simulate("click");
+    expect(searchCallback.callCount).to.equal(1);
```

Comment:
Should `mount` be avoided as much as possible, or is it just speed of running tests?

Reply:
I think there are times when you want to test things that can't be tested using a shallow render so it's fine to mount. This is convenience more than anything so it's fine if you don't want to switch it. Usually I find that it's the tests with complex behavior and API calls that slow down tests rather than searching in the DOM.

Reply:
Nah I just wanted to know in general, the internet is very divided on when/whether to use mount